tests/other/no-exceptions/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+#===============================================================================
+# Copyright 2016-2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+if (NOT DNNL_BUILD_EXAMPLES)
+    return()
+endif()
+
+# propagate EXAMPLE specific flags
+append(CMAKE_C_FLAGS "${CMAKE_EXAMPLE_CCXX_FLAGS}")
+append(CMAKE_CXX_FLAGS "${CMAKE_EXAMPLE_CCXX_FLAGS}")
+
+# propagate sanitizer flags
+append(CMAKE_C_FLAGS "${CMAKE_CCXX_SANITIZER_FLAGS}")
+append(CMAKE_CXX_FLAGS "${CMAKE_CCXX_SANITIZER_FLAGS}")
+
+# propagate nowarn flags
+append(CMAKE_C_FLAGS "${CMAKE_CCXX_NOWARN_FLAGS}")
+append(CMAKE_CXX_FLAGS "${CMAKE_CCXX_NOWARN_FLAGS}")
```

Comment:
It seems these flags should be (and actually are) propagated in `tests/CMakeLists.txt`. Can we remove the above `append` calls from here?